packages/lodestar/src/chain/chain.ts

```diff
@@ -362,6 +395,30 @@ export class BeaconChain implements IBeaconChain {
     });
   }
 
+  /**
+   * Seeds the fork choice with an anchor state.
+   * This state is set as the finalized state.
+   */
+  private async initForkChoiceFromWeakSubjectivityState(anchorState: TreeBacked<BeaconState>): Promise<void> {
+    const blockHeader = anchorState.latestBlockHeader.clone() as BeaconBlockHeader;
+    blockHeader.stateRoot = anchorState.hashTreeRoot();
+    const blockRoot = this.config.types.BeaconBlockHeader.hashTreeRoot(blockHeader);
+    const epoch = computeEpochAtSlot(this.config, anchorState.slot);
+    const blockCheckpoint = {
+      root: blockRoot,
+      // TODO: is it possible to be `epoch + n` where n >= 2?
+      epoch: anchorState.slot === computeStartSlotAtEpoch(this.config, epoch) ? epoch : epoch + 1,
+    };
+    this.forkChoice.addBlock({
+      slot: anchorState.slot,
+      blockRoot,
+      stateRoot: this.config.types.BeaconState.hashTreeRoot(anchorState),
+      parentRoot: anchorState.latestBlockHeader.parentRoot.valueOf() as Uint8Array,
+      justifiedCheckpoint: blockCheckpoint,
```

Comment:
I believe these checkpoints should be state.currentJustifiedCheckpoint and state.finalizedCheckpoint.

Reply:
Oh, I see... I guess there's two interpretations of 'what to do', both with their own issues:
- set this state.latestBlockHeader as the finalized/justified checkpoint, in which case we don't actually have a valid block to add to the fork choice (we don't have a block where this state is the one thats finalized)
or
- set this state.latestBlockHeader as the block in the fork choice, in which case we don't have the blocks from the finalized/justified checkpoint up to the lastestBlock

These do different things, and I think we should figure out which is the 'correct' one.
My current thought is that the first interpretation is correct. In which case, what you have here for the checkpoints is correct. I think we will need an anchor block as well as a state in order to restore the chain.

This aligns with the needs during genesis as well. Its just that for genesis, we can compute the genesis block, and don't need to provide it separately.

Reply:
if we go with "`state.latestBlockHeader as the finalized/justified checkpoint`" then I think we don't need anchor block since we have `stateRoot` as well, so that's enough to add to forkchoice

the concern for this approach is which epoch for the finalized/justified checkpoint is correct? in a lot of cases its `computeEpochAtSlot()` but it's not always be like that.

I wonder if one of the criteria to select a weak subjectivity state is it should not have a skipped slot at start of that epoch ...

Reply:
ain't wrong finalized checkpoint mess with peer status exchange for peer that check if we have proper finalized checkpoint?

Reply:
yeah that's why I got goodbye requests a lot using `computeEpochAtSlot()`. We need to make sure we have correct finalized checkpoint.